designer/server/src/test/scala/pl/touk/nussknacker/ui/api/MigrationApiHttpServiceBusinessSpec.scala

```diff
@@ -0,0 +1,142 @@
+import pl.touk.nussknacker.restmodel.validation.ValidationResults.{
+  NodeValidationError,
+  NodeValidationErrorType,
+  ValidationResult
+}
+import pl.touk.nussknacker.test.base.it.{NuItTest, WithRichConfigScenarioHelper}
+import pl.touk.nussknacker.test.config.WithRichDesignerConfig.TestCategory.Category1
+import pl.touk.nussknacker.test.config.{WithMockableDeploymentManager, WithRichDesignerConfig}
+import pl.touk.nussknacker.test.processes.WithRichScenarioActivitySpecAsserts
+import pl.touk.nussknacker.test.{NuRestAssureExtensions, NuRestAssureMatchers, RestAssuredVerboseLogging}
+import pl.touk.nussknacker.ui.process.marshall.CanonicalProcessConverter
+
+class MigrationApiHttpServiceBusinessSpec
+    extends AnyFreeSpecLike
+    with NuItTest
+    with WithRichDesignerConfig
+    with WithRichConfigScenarioHelper
+    with WithMockableDeploymentManager
+    with WithRichScenarioActivitySpecAsserts
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging {
+
+  "The endpoint for scenario migration between environments should" - {
+    "migrate scenario and add update comment" in {
+      given()
+        .applicationState(
+          createSavedScenario(exampleScenario, Category1)
+        )
+        .when()
+        .basicAuthAllPermUser()
+        .jsonBody(validRequestData)
+        .post(s"$nuDesignerHttpAddress/api/migrate")
+        .Then()
+        .statusCode(200)
+        .verifyCommentExists(exampleProcessName.value, "Scenario migrated from DEV by allpermuser")
+    }
+    "fail when scenario name contains illegal character(s)" in {
+      given()
+        .when()
+        .basicAuthAllPermUser()
+        .jsonBody(requestDataWithInvalidScenarioName)
+        .post(s"$nuDesignerHttpAddress/api/migrate")
+        .Then()
+        .statusCode(400)
+        .equalsPlainBody(
+          "Cannot migrate, following errors occurred: Invalid scenario name #test. Only digits, letters, underscore (_), hyphen (-) and space in the middle are allowed"
+        )
+    }
+    "fail when scenario is archived on target environment" in {
+      given()
+        .applicationState(
+          createArchivedExampleScenario(exampleProcessName, Category1)
+        )
+        .when()
+        .basicAuthAllPermUser()
+        .jsonBody(validRequestData)
+        .post(s"$nuDesignerHttpAddress/api/migrate")
+        .Then()
+        .statusCode(400)
+        .equalsPlainBody(
+          s"Cannot migrate, scenario ${exampleProcessName.value} is archived on test. You have to unarchive scenario on test in order to migrate."
+        )
+    }
+    "migrate fragment" in {
+      given()
+        .applicationState(
+          createSavedScenario(validFragment, Category1)
+        )
+        .when()
+        .basicAuthAllPermUser()
+        .jsonBody(validRequestDataForFragment)
+        .post(s"$nuDesignerHttpAddress/api/migrate")
+        .Then()
+        .statusCode(200)
```

Comment:
status code 200 and empty response body means that it works?

Reply:
I think I wanted to check that _it does not fail_.